src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.Changed.cs

```diff
@@ -118,4 +118,128 @@ public static void FileSystemWatcher_Changed_FileInNestedDirectory()
+                // Attach the FSW to the existing structure
+                watcher.Path = Path.GetFullPath(dir.Path);
+                watcher.Filter = "*";
+                watcher.NotifyFilter = NotifyFilters.Attributes | NotifyFilters.CreationTime | NotifyFilters.DirectoryName | NotifyFilters.FileName | NotifyFilters.LastAccess | NotifyFilters.LastWrite | NotifyFilters.Security | NotifyFilters.Size;
+                watcher.IncludeSubdirectories = true;
+                watcher.EnableRaisingEvents = true;
+
+                // Change the nested file and verify we get the changed event
+                file.Write(bt, 0, bt.Length);
+                file.Flush();
+            }
+
+            Utility.ExpectEvent(eventOccured, "file changed", 10000);
+
+            // Stop the FSW
+            watcher.EnableRaisingEvents = false;
+
+            using (FileStream file2 = File.Create(Path.Combine(dir3.Path, "testfile2.txt")))
+            {
+                // Start the FSW, write to the file, expect to pick up the write
+                watcher.EnableRaisingEvents = true;
+                file2.Write(bt, 0, bt.Length);
+                file2.Flush();
+            }
+
+            Utility.ExpectEvent(eventOccured, "second file change");
+        }
+    }
+
+    [Fact, ActiveIssue("1477", PlatformID.Windows)]
+    public static void FileSystemWatcher_Changed_SymLinkFileDoesntFireEvent()
+    {
+        using (var dir = Utility.CreateTestDirectory())
+        using (var file = Utility.CreateTestFile(Path.GetTempFileName()))
+        using (var watcher = new FileSystemWatcher())
+        {
+            // Create the symlink first
+            Utility.CreateSymLink(Path.GetFullPath(file.Path), Path.Combine(dir.Path, "link"), false);
+
+            AutoResetEvent are = Utility.WatchForEvents(watcher, WatcherChangeTypes.Changed);
+
+            // Setup the watcher
+            watcher.Path = Path.GetFullPath(dir.Path);
+            watcher.Filter = "*";
+            watcher.NotifyFilter = NotifyFilters.FileName | NotifyFilters.Size;
+            watcher.EnableRaisingEvents = true;
+
+            // Changing the temp file should not fire an event through the symlink
+            byte[] bt = new byte[4096];
+            file.Write(bt, 0, bt.Length);
+            file.Flush();
+            Utility.ExpectNoEvent(are, "symlink'd file change");
+        }
+    }
+
+    [Fact, ActiveIssue("1477", PlatformID.Windows)]
+    public static void FileSystemWatcher_Changed_SymLinkFolderDoesntFireEvent()
+    {
+        using (var dir = Utility.CreateTestDirectory())
+        using (var tempDir = Utility.CreateTestDirectory(Path.Combine(Path.GetTempPath(), "FooBar")))
+        using (var file = Utility.CreateTestFile(Path.Combine(tempDir.Path, "Test")))
+        using (var watcher = new FileSystemWatcher())
+        {
+            // Create the symlink first
+            Utility.CreateSymLink(Path.GetFullPath(file.Path), Path.Combine(dir.Path, "link"), false);
```

Comment:
I'm probably just missing it, but I'm not understanding what this test is meant to be testing.  The watcher is set up to watch dir.Path, but the symlink is being created for a file in a temporary directory that's unrelated to dir.Path.  From the test's name, I was expecting to see this test doing the same thing as the previous test, expect with a symlink to a directory rather than to a file.